lib/types.ts

```diff
@@ -46,7 +48,7 @@ export interface PluginResponse {
 
 export interface ScannedProjectCustom {
   packageManager: string; // actually SupportedPackageManagers; in the CLI
-  depTree: DepTree;
+  depTree: DepTree | PkgTree;
```

Comment:
could we have the type here always be just ```DepTree```? if ```PkgTree``` has discrepancies then adapt them to this format? better to have the return type deterministic if possible, I think?

Reply:
The PkgTree is pretty much DepTree but with all fields optional... (And missing `docker`, `targetOS`...) I find it really hard to adapt it unless I force-cast and skip all compiler checks

Reply:
is depTree a must? I mean going forward if we start adding non-dep-tree stuff the depTree wont have a lot of meaning

Reply:
I think it's a must from the CLI's perspective?

Reply:
true. I think we would need to change it to be optional in the cli-interface as well. but maybe not part of this commit

Reply:
I'm still troubled by this ☹️
maybe we should merge these two types? or is this insane? I have no idea.
maybe even just add a comment explaining what each of these two things mean for a later refactor to be made easier with more context?